src/Avalonia.FuncUI/DSL/Base/StyledElement.fs

```diff
@@ -5,11 +5,165 @@ open Avalonia.Controls.Primitives
 [<AutoOpen>]
 module StyledElement =  
     open Avalonia
+    open Avalonia.FuncUI
     open Avalonia.FuncUI.Types
     open Avalonia.FuncUI.Builder
     open Avalonia.Styling
+    open Avalonia.LogicalTree
+    open System.Threading
+    open System
+
+    module internal ClassesInternals =
+        open System.Linq
+
+        /// pseudoclasse is classe beginning with a ':' character.
+        let isPseudoClass (s: string) = s.StartsWith(":")
+
+        /// Update `Classes`'s standard classes with new values. 
+        let patchStandardClasses (classes: Classes) (newValues: string seq) =
```

Comment:
@SilkyFowl could you add a comment here explaining what the problem is any why it makes sense to write a custom differ here? 

(I get it, but would be nice to have it in the sources)

Reply:
I've made the comments more detailed. What do you think? 
Additionally, I'm considering adding a process to verify that pseudoclasses are not included in the binding.

```fs
    /// consider: detail explain comment should be written here?
    module internal ClassesInternals =
        open System.Linq
        
        /// ...

        let validateStandardClasses (classes: string seq) =
            if classes |> Seq.exists isPseudoClass then
                failwith "Pseudoclasses may only setting by the control's protected PseudoClasses property itself."

    type StyledElement with
        /// ...

        /// todo: add comment explaining Pseudoclasses 
        static member classes<'t when 't :> StyledElement>(value: Classes) : IAttr<'t> =
            ClassesInternals.validateStandardClasses value

            let getter: ('t -> Classes) = (fun control -> control.Classes)
            /// ...

        /// todo: add comment explaining Pseudoclasses 
        static member classes<'t when 't :> StyledElement>(classes: string list) : IAttr<'t> =
            ClassesInternals.validateStandardClasses classes
            /// ...
```

Reply:
Looks good to me, even tho I was more thinking about explaining why there is special logic for diffing/patching style classes. But we can add that later.